metric/system/process/process.go

```diff
@@ -265,6 +279,7 @@ func (procStats *Stats) pidFill(pid int, filter bool) (ProcState, bool, error) {
 		if !errors.Is(err, NonFatalErr{}) {
 			return status, true, fmt.Errorf("FillPidMetrics: %w", err)
 		}
+		wrappedErr = errors.Join(wrappedErr, fmt.Errorf("non-fatal error fetching PID metrics for %d, metrics are valid, but partial: %w", pid, err))
```

Comment:
Same here, the Debugf on the following line can just print the error string.

Reply:
Also add a unit test that doing `fmt.Errorf()` with `%w` when the wrapped error is of type NonFatalError still satisifies an `errors.Is(err, NonFatalErr)` check.

There are no tests of the `NonFatalErr` type and I see an Unwrap method had to be added probably for this reason.

Reply:
> Also add a unit test that doing fmt.Errorf() with %w when the wrapped error is of type NonFatalError still satisifies an errors.Is(err, NonFatalErr) check.

Sure.

Reply:
@cmacknz I've added some test cases covering hierarchy of errors, non fatal and fatal scenarios.